src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas1/Cas1BookingEmailServiceTest.kt

```diff
@@ -101,207 +138,181 @@ class Cas1BookingEmailServiceTest {
-      "bookingUrl" to "http://frontend/premises/${premises.id}/bookings/${booking.id}",
-      "crn" to CRN,
-      "startDate" to "2023-02-01",
-      "endDate" to "2023-02-14",
-      "lengthStay" to 2,
-      "lengthStayUnit" to "weeks",
-    )
-
-    mockEmailNotificationService.assertEmailRequested(
-      APPLICANT_EMAIL,
-      notifyConfig.templates.bookingMade,
-      personalisation,
-    )
-
-    mockEmailNotificationService.assertEmailRequested(
-      PREMISES_EMAIL,
-      notifyConfig.templates.bookingMadePremises,
-      personalisation,
-    )
-  }
-
-  @Test
-  fun `bookingMade sends email to applicant and premises email addresses when defined, when length of stay not whole number of weeks`() {
-    val applicant = UserEntityFactory()
-      .withUnitTestControlProbationRegion()
-      .withEmail(APPLICANT_EMAIL)
-      .produce()
-
-    val (application, booking) = createApplicationAndBooking(
-      applicant,
-      premises,
-      arrivalDate = LocalDate.of(2023, 2, 22),
-      departureDate = LocalDate.of(2023, 2, 27),
-    )
-
-    service.bookingMade(application, booking)
-
-    mockEmailNotificationService.assertEmailRequestCount(2)
-
-    val expectedPersonalisation = mapOf(
-      "lengthStay" to 6,
-      "lengthStayUnit" to "days",
-    )
-
-    mockEmailNotificationService.assertEmailRequested(
-      APPLICANT_EMAIL,
-      notifyConfig.templates.bookingMade,
-      expectedPersonalisation,
-    )
-
-    mockEmailNotificationService.assertEmailRequested(
-      PREMISES_EMAIL,
-      notifyConfig.templates.bookingMadePremises,
-      expectedPersonalisation,
-    )
-  }
-
-  @Test
-  fun `bookingWithdrawn sends email to applicant, premises and CRU if emails are defined`() {
-    val applicant = UserEntityFactory()
-      .withUnitTestControlProbationRegion()
-      .withEmail(APPLICANT_EMAIL)
-      .produce()
+      )
+
+      mockEmailNotificationService.assertEmailRequested(
+        APPLICANT_EMAIL,
+        notifyConfig.templates.bookingMade,
+        personalisation,
+      )
+
+      mockEmailNotificationService.assertEmailRequested(
+        PREMISES_EMAIL,
+        notifyConfig.templates.bookingMadePremises,
+        personalisation,
+      )
+    }
+
+    @Test
+    fun `bookingMade sends email to applicant and premises email addresses when defined, when length of stay not whole number of weeks`() {
+      val applicant = UserEntityFactory()
+        .withUnitTestControlProbationRegion()
+        .withEmail(APPLICANT_EMAIL)
+        .produce()
+
+      val (application, booking) = createApplicationAndBooking(
+        applicant,
+        premises,
+        arrivalDate = LocalDate.of(2023, 2, 22),
+        departureDate = LocalDate.of(2023, 2, 27),
+      )
+
+      service.bookingMade(application, booking)
+
+      mockEmailNotificationService.assertEmailRequestCount(2)
+
+      val expectedPersonalisation = mapOf(
+        "lengthStay" to 6,
+        "lengthStayUnit" to "days",
+      )
```

Comment:
This appears to be the logic:

```
"lengthStay" to if (lengthOfStayWeeksWholeNumber) lengthOfStayWeeks.toInt() else lengthOfStayDays,
"lengthStayUnit" to if (lengthOfStayWeeksWholeNumber) "weeks" else "days",
```

So unless it's divisible by 7 we'll always get the complete day count (86 days)

On my domain event PR i've added a function to present durations as week / days, so this would be a good candidate to use it.

Reply:
you read my mind!